test/integration/connect/envoy/case-api-gateway-http-splitter-targets/setup.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -euo pipefail
+
+upsert_config_entry primary '
+kind = "api-gateway"
+name = "api-gateway"
+listeners = [
+  {
+    name = "listener-one"
+    port = 9999
+    protocol = "http"
+  }
+]
+'
+
+upsert_config_entry primary '
+Kind      = "proxy-defaults"
+Name      = "global"
+Config {
+  protocol = "http"
+}
+'
+
+upsert_config_entry primary '
+kind = "http-route"
+name = "api-gateway-route-one"
+rules = [
+  {
+    services = [
+      {
+        name = "splitter-one"
+      }
+    ]
+  }
+]
+parents = [
+  {
+    name = "api-gateway"
+    sectionName = "listener-one"
+  }
```

Comment:
Does `kind` default to `"api-gateway"` here?

My belief was "no" coming into this PR, based on
https://github.com/hashicorp/consul/blob/15d2684ecc5c97036d946c8e8f4578446a56b3e8/agent/structs/config_entry_routes.go#L134-L142

Reply:
It get's normalized here if unset: https://github.com/hashicorp/consul/blob/15d2684ecc5c97036d946c8e8f4578446a56b3e8/agent/structs/config_entry_routes.go#L80-L84